src/InnoSetup/Environment.iss

```diff
@@ -345,8 +345,16 @@ begin
   { If there is a release archive to download, IDFZIPFileName and IDFZIPFileVersion will be set.
     See GetIDFZIPFileVersion function.
   }
+  NeedToClone := False;
 
-  if IDFZIPFileName <> '' then
+  if WildCardMatch(IDFDownloadVersion, 'release*') then
+  begin
+    { Instead of downloading .zip archive and then fast forward, performing clone for the release branches }
+    NeedToClone := True;
+    Log('Performing full clone for the release branch.');
+  end;
+  
+  if (not NeedToClone) and (IDFZIPFileName <> '') then
   begin
     if IDFZIPFileVersion <> IDFDownloadVersion then
```

Comment:
Seems like this whole code block will never be executed now. Instead of adding new code, could you simply remove the existing code which handles the fast-forwarding logic?

(And all the supporting functions, for example GitRepoDissociate and others)

Reply:
We should still use archive download for release versions (tags), I think the change is only to do `git clone` for release branches — instead of downloading a nearest release version and then fetching the changes between it and the tip of the release branch.